numpy/random/mtrand.pyx

```diff
@@ -4060,6 +4060,32 @@ cdef class RandomState:
             shape = [size]
         else:
             shape = size
+                  
+        if size is None and len(mean.shape) != 1:
+            if len(mean.shape) != len(cov.shape) - 1:
+                raise ValueError("mean and cov must have same size")
+            for m, c in zip(mean.shape, cov.shape):
+                if m != c:
+                    raise ValueError("mean and cov must have same size")
+            if cov.shape[len(cov.shape) - 1] != cov.shape[len(cov.shape) - 2]:
+                raise ValueError("cov must be square")
+            if mean.shape[len(mean.shape) - 1] != cov.shape[len(cov.shape) - 1]:
+                raise ValueError("mean and cov must have same length")
+
+            result = np.zeros(mean.shape)
+            cov.reshape(-1, *cov.shape[-2:])
+            final_shape = list(shape[:])
+            final_shape.append(mean.shape[len(mean.shape) - 1])
+            for i in range(cov.shape[0]):
```

Comment:
Does this approach lead you to perform the SVD multiple times on the same covariance method?  I thought about this problem once and felt that a good solution would need to perform the minimal number of SVDs.

Reply:
I think you could compute the SVD for the set of distinct cov matrices and then loop over these SVDs. This isn't the most efficient since it requires a bit more memory, but will avoid repeatedly SVDing the same covariance matrix.